dask_searchcv/tests/test_model_selection.py

```diff
@@ -620,3 +620,63 @@ def test_scheduler_param_distributed(loop):
 def test_scheduler_param_bad():
     with pytest.raises(ValueError):
         _normalize_scheduler('threeding', 4)
+
+
+class CacheCVLike(CVCache):
+    called = 0
+    def __init__(self, splits=None, pairwise=False, cache=True):
+        CVCache.__init__(self, splits=splits, pairwise=pairwise, cache=cache)
+
+    def extract(self, X, y, n, is_x=True, is_train=True):
+        CacheCVLike.called += 1
+        X, y = self.make_data()
+        if is_x:
+            return X
+        else:
+            return y
+
+    def make_data(self):
+        return make_classification(n_samples=100,
+                                   n_features=10,
+                                   random_state=0)
+
+
+class CacheCVBad(CacheCVLike):
+
+    extract = CVCache.extract
+
+
+@pytest.mark.parametrize('cls, refit, should_pass', [
+    (cls, refit, sp1 and sp2)
+    for cls, sp1 in zip((CacheCVLike, CacheCVBad), (True, False))
+    for refit, sp2 in zip((True, False, 'sample'), (False, False, True))
```

Comment:
Could you expand this? I'm having trouble following it :)

IIUC, it's

```python
(CacheCVLike, True, False),
(CacheCVLike, False, False),
(CacheCVLike, 'sample', True),

(CacheCVBad, True, False),
(CacheCVBad, False, False),
(CacheCVBad, 'sample', False),
```

?

Reply:
For these test arguments:
```
(CacheCVBad, True, False),
(CacheCVBad, False, False),
(CacheCVBad, 'sample', False),

```
 * `CacheCVBad` should always fail (the `False` as the last item as each tuple).  It has the same `CVCache.extract` as its `extract` method and thus cannot do the sampling logic.  

And for these ones:

```
(CacheCVLike, True, False),
(CacheCVLike, False, False),
(CacheCVLike, 'sample', True),
```

* An instance of `CacheCVLike` is an okay `cache_cv` argument as long as `X` or `X, y` are passed as `refit` 

TODO for me on this one:
 * [ ] I'll add notes in the docstring mentioning those ideas.  
 * [ ] Probably better to just type out the args than do that double for loop
 * [ ] Make some docstring examples demonstrating usage of `cache_cv` / `refit` options

Onoe thing, @TomAugspurger , do you think I should just change `CVCache` in `methods.py` to take `splits=None` or leave `splits` as required?  Making it a keyword argument would let me remove the `__init__` from `CVCacheLike`